examples/mnist/train.cpp

```diff
@@ -79,10 +79,12 @@ static void train_lenet(const std::string& data_dir_path) {
 
   progress_display disp(static_cast<unsigned long>(train_images.size()));
   timer t;
-  int minibatch_size = 10;
+  int minibatch_size = 16;
   int num_epochs     = 30;
 
-  optimizer.alpha *= static_cast<tiny_dnn::float_t>(std::sqrt(minibatch_size));
+  optimizer.alpha *=
+    std::min(tiny_dnn::float_t(4),
+             static_cast<tiny_dnn::float_t>(std::sqrt(minibatch_size)));
```

Comment:
@beru In numbers, say we pick minibatch_size = 16. The default start learning rate set by tiny-dnn in adagrad is alpha=0.01. We end up with alpha = 0.04 with the HEAD and it the algorithm takes it well: convergence speeds up without stalling it. But alpha bigger than that is making the algorithm to stall. 

Regarding the casting, yes, either way suppresses the warning although the scaling value would be different. The way you suggest would make the alpha scaling more conservative because the sqrt would be floored.  Either way is fine with me. Let me know.